src/snowcli/cli/snowpark/commands.py

```diff
@@ -206,18 +211,61 @@ def deploy(
     return CollectionResult(deploy_status)
 
 
+def _check_if_all_defined_integrations_exists(
+    om: ObjectManager, functions: List[Dict], procedures: List[Dict]
+):
+    existing_integrations = [
+        i["name"].lower()
+        for i in om.show(object_type="integration", cursor_class=DictCursor, like=None)
+        if i["type"] == "EXTERNAL_ACCESS"
+    ]
+    for object_definition in [*functions, *procedures]:
+        external_access_integrations = [
+            s.lower() for s in object_definition.get("external_access_integrations", [])
+        ]
+        secrets = [s.lower() for s in object_definition.get("secrets", [])]
+
+        if not external_access_integrations and secrets:
+            raise SecretsWithoutExternalAccessIntegrationError()
+
+        if any(i not in existing_integrations for i in external_access_integrations):
+            raise ClickException(
+                "One of defined external access integrations does not exist in Snowflake."
+            )
+
+
+def _check_if_any_object_exist_already(fm, functions, pm, procedures, replace):
+    existing_functions = fm.get_existing_objects()
+    existing_procedures = pm.get_existing_objects()
+    existing_objects = []
+    object_state = {
+        "function": (functions, existing_functions),
+        "procedure": (procedures, existing_procedures),
+    }
+    for object_type, (objects, existing) in object_state.items():
+        for obj in objects:
+            full_object_identifier = object_to_signature(obj)
+            if full_object_identifier in existing:
+                existing_objects.append(f"{object_type}: {full_object_identifier}")
+    if any(existing_objects) and not replace:
+        existing = "\n".join(existing_objects)
+        raise ClickException(
+            f"Following object already exists. Consider using --replace.\n {existing}"
+        )
```

Comment:
It is so hard to understand what is going here 😄

Reply:
I split this logic, hopefully it will be simpler